library/src/main/java/org/xmtp/android/library/Conversation.kt

```diff
@@ -28,154 +29,130 @@ sealed class Conversation {
-                    ),
-            ).build()
-
-            is Group -> throw XMTPException("Groups do not support topics")
+            is V1 -> throw XMTPException("Only supported for V3")
+            is V2 -> throw XMTPException("Only supported for V3")
+            is Group -> group.members()
+            is Dm -> dm.members()
         }
     }
 
-    fun decode(envelope: Envelope, message: MessageV3? = null): DecodedMessage {
+    suspend fun updateConsentState(state: ConsentState) {
         return when (this) {
-            is V1 -> conversationV1.decode(envelope)
-            is V2 -> conversationV2.decodeEnvelope(envelope)
-            is Group -> message?.decode() ?: throw XMTPException("Groups require message be passed")
-        }
-    }
-
-    fun decodeOrNull(envelope: Envelope): DecodedMessage? {
-        return try {
-            decode(envelope)
-        } catch (e: Exception) {
-            Log.d("CONVERSATION", "discarding message that failed to decode", e)
-            null
-        }
-    }
-
-    fun <T> prepareMessage(content: T, options: SendOptions? = null): PreparedMessage {
-        return when (this) {
-            is V1 -> {
-                conversationV1.prepareMessage(content = content, options = options)
-            }
-
-            is V2 -> {
-                conversationV2.prepareMessage(content = content, options = options)
-            }
-
-            is Group -> throw XMTPException("Groups do not support prepared messages") // We return a encoded content not a preparedmessage which requires a envelope
+            is V1 -> throw XMTPException("Only supported for V3")
+            is V2 -> throw XMTPException("Only supported for V3")
+            is Group -> group.updateConsentState(state)
+            is Dm -> dm.updateConsentState(state)
         }
     }
 
-    fun prepareMessage(
-        encodedContent: EncodedContent,
-        options: SendOptions? = null,
-    ): PreparedMessage {
+    suspend fun consentState(): ConsentState {
         return when (this) {
-            is V1 -> {
-                conversationV1.prepareMessage(encodedContent = encodedContent, options = options)
-            }
-
-            is V2 -> {
-                conversationV2.prepareMessage(encodedContent = encodedContent, options = options)
-            }
-
-            is Group -> throw XMTPException("Groups do not support prepared messages") // We return a encoded content not a preparedmessage which requires a envelope
+            is V1 -> conversationV1.client.contacts.consentList.state(address = peerAddress)
+            is V2 -> conversationV2.client.contacts.consentList.state(address = peerAddress)
+            is Group -> group.consentState()
+            is Dm -> dm.consentState()
         }
     }
 
-    suspend fun send(prepared: PreparedMessage): String {
+    suspend fun <T> prepareMessageV3(content: T, options: SendOptions? = null): String {
```

Comment:
Ya it's a bit unfortunate. I tried several different hacks to try and make it work for both V3 and V2 but since V2 depends so heavily on the envelope it would be a break change. I'm cool with not exposing this in the interface and just cutting the old one over eventually. Integrators will likely call these methods directly on the group or dm anyways.